lib/avromatic/model_registry.rb

```diff
@@ -0,0 +1,43 @@
+module Avromatic
+  # The ModelRegistry class is used to store and fetch nested models by
+  # their fullname. An optional namespace prefix can be removed from the full
+  # name that is used to store and fetch models.
+  class ModelRegistry
+
+    def initialize(remove_namespace_prefix: nil)
+      @prefix = remove_namespace_prefix
+      @hash = Hash.new
+    end
+
+    def [](fullname)
+      @hash.fetch(fullname)
+    end
+
+    def register(model)
+      raise 'models with a key schema are not supported' if model.key_avro_schema
+      name = model.avro_schema.fullname
+      name = remove_prefix(name) if @prefix
+      @hash[name] = model
+    end
+
+    def model?(fullname)
```

Comment:
Calling this `registered?` would make the naming more consistent with `register` and might make code at calling sites a bit more readable.